src/components/widgets/BaseInput.js

```diff
@@ -4,6 +4,10 @@ import PropTypes from "prop-types";
 function BaseInput(props) {
   // Note: since React 15.2.0 we can't forward unknown element attributes, so we
   // exclude the "options" and "schema" ones here.
+  if (!props.id) {
+    console.log("No id for", props);
+    throw new Error(`no id for props ${JSON.stringify(props)}`);
+  }
```

Comment:
I think it would be better for this code not to come between the comment and the code the comment describes.